### ISSUE TITLE
Events not firing for Upload Component

### ISSUE THREAD

**Issue by PaitoAnderson** [telerik/kendo-react#244]
## I'm submitting a...
* Bug report

## Current behavior
After selecting and successfully uploading a file only the `onBeforeUpload` event has fired.

## Expected behavior
I would have expected `onAdd`, `onProgress`, and maybe `onStatusChange`? events to have fired after a file is selected and successfully uploaded.

## Minimal reproduction of the problem with instructions
```js
import React from 'react';
import ReactDOM from 'react-dom';
import { Upload } from '@progress/kendo-react-upload';

class App extends React.Component {
    render() {
        return (
            <Upload
                batch={false}
                multiple={true}
                defaultFiles={[]}
                withCredentials={false}
                saveUrl={'https://demos.telerik.com/kendo-ui/service-v4/upload/save'}
                removeUrl={'https://demos.telerik.com/kendo-ui/service-v4/upload/remove'}
                onStatusChange={(e) => console.log('onStatusChange', e)}
                onProgress={(e) => console.log('onProgress', e)}
                onAdd={(e) => console.log('onAdd', e)}
                onBeforeUpload={(e) => console.log('onBeforeUpload', e)}
                onBeforeRemove={(e) => console.log('onBeforeRemove', e)}
                onRemove={(e) => console.log('onRemove', e)}
            />
        );
    }
}

ReactDOM.render(
    <App />,
    document.querySelector('my-app')
);
```

https://7udwxx.run.stackblitz.io/

## What is the motivation or use case for changing the behavior?
I'd like a callback after a file is successfully uploaded to hide the uploader and progress with the workflow.

## Environment

Package versions:
```
├── @progress/kendo-data-query@1.5.1
├── @progress/kendo-drawing@1.5.11
├── @progress/kendo-react-buttons@2.10.0
├── @progress/kendo-react-charts@2.10.0
├── @progress/kendo-react-dateinputs@2.10.0
├── @progress/kendo-react-dialogs@2.10.0
├── @progress/kendo-react-dropdowns@2.10.0
├── @progress/kendo-react-editor@2.10.0
├── @progress/kendo-react-grid@2.10.0
├── @progress/kendo-react-inputs@2.10.0
├── @progress/kendo-react-intl@2.10.0
├── @progress/kendo-react-layout@2.10.0
├── @progress/kendo-react-pdf@2.10.0
├── @progress/kendo-react-upload@2.10.0
├── @progress/kendo-theme-bootstrap@3.3.1
```

Browser:
- Chrome (desktop) 74.0.3729.131

System:
- TypeScript version: 3.2.2
- Node version: v11.10.0 & v6.9.0
- Platform:  macOS

**Comment by Xizario** [comment]
In the current implementation these events are working for controlled mode only (when `files` are set instead of `defaultFiles`)
https://www.telerik.com/kendo-react-ui/components/upload/modes/#toc-controlled-mode

However I see some benefits of providing the events at all cases as they tare working in other form components. Or at-least we need to improve the docs page.

**Comment by vladimirivanoviliev** [comment]
It appears that having these events in uncontrolled is valid scenario and we will add them in next official release.

**Comment by Xizario** [comment]
Fixed in 2.11.0
